fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/ConstantValue.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * LiteralValue.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2022 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.query.plan.cascades.values;
+
+import com.apple.foundationdb.annotation.API;
+import com.apple.foundationdb.annotation.SpotBugsSuppressWarnings;
+import com.apple.foundationdb.record.EvaluationContext;
+import com.apple.foundationdb.record.ObjectPlanHash;
+import com.apple.foundationdb.record.PlanHashable;
+import com.apple.foundationdb.record.provider.foundationdb.FDBRecordStoreBase;
+import com.apple.foundationdb.record.query.plan.cascades.AliasMap;
+import com.apple.foundationdb.record.query.plan.cascades.CorrelationIdentifier;
+import com.apple.foundationdb.record.query.plan.cascades.Formatter;
+import com.apple.foundationdb.record.query.plan.cascades.typing.Type;
+import com.google.protobuf.Message;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * A wrapper around a constant.
```

Comment:
Needs some expansion of why / how this is used. It isn't immediately obvious what folding constants into one of these wrappers buys. What is the emergent behavior from this wrapper?

Reply:
I can take that back out as it is as of today not actually used in any useful ways. It will allow the simplification logic to fold a `Value` tree into a simpler one where this _nested_ `Value` is considered extractable to a higher level, i.e. outside of the current query block into a higher one, or, if no further correlations are present, into some sort of a all-encompassing NLJN of a `SELECT inner FROM VALUES(all the constants as columns) v, ...thequery... as inner` where all the constants are now values of the shape `v.something`. This can help us with literal extractions, etc.

Reply:
Should I take it back out?